### PR TITLE
press summary for promotion awards / token based authentication 

### PR DESCRIPTION
**Pull request by dreamfall**
add token

### CODE REVIEW

File: spec/mailers/users/winners_press_release_spec.rb [hunk 1]
```diff
@@ -33,7 +34,7 @@
     it "renders the body" do
       expect(mail.body.encoded).to match(user.decorate.full_name)
       expect(mail.body.encoded).to have_link("Review Press Comment",
-                                             href: users_form_answer_press_summary_url(form_answer))
+                                             href: users_form_answer_press_summary_url(form_answer, token: press_release.token))
```
Comment (matthewford) [inline spec/mailers/users/winners_press_release_spec.rb]: Line is too long. [128/100]

File: app/models/press_summary.rb [hunk 2]
```diff
@@ -2,11 +2,17 @@ class PressSummary < ActiveRecord::Base
...
   validates :form_answer, :body, presence: true
+  validates :first_name, :last_name, :email, :phone_number, presence: true, if: :reviewed_by_user?
 
+  before_create :set_token
   after_save :notify_applicant
 
   private
 
+  def set_token
+    self.token = SecureRandom.base64(24)
```
Comment (jaleszek) [inline app/models/press_summary.rb]: do we need any validations constraints for token ?

File: app/controllers/users/press_summaries_controller.rb [hunk 3]
```diff
@@ -1,31 +1,34 @@
 class Users::PressSummariesController < Users::BaseController
```
Comment (jaleszek) [inline app/controllers/users/press_summaries_controller.rb]: Maybe just inherit it from ApplicationsController ?

Comment (jaleszek) [inline app/controllers/users/press_summaries_controller.rb]: nope, ignore that

File: app/controllers/users/press_summaries_controller.rb [hunk 4]
```diff
@@ -1,31 +1,34 @@
...
-      redirect_to action: :show
+      redirect_to action: :show, token: params[:token]
     else
       render :show
     end
   end
 
+  def update_acceptance
+    if params[:form_answer][:accepted] == "true"
+      form_answer.update_column(:accepted, true)
```
Comment (jaleszek) [inline app/controllers/users/press_summaries_controller.rb]: Skipping validations is necessary here ?

Comment (dreamfall) [inline app/controllers/users/press_summaries_controller.rb]: @jaleszek no. But there's no point in validating form answer after the flag is changed.

**Comment by jaleszek** [comment]
:+1:  looks good, just let me know about your thoughts about validating token

**Pull request merged**